### ISSUE TITLE
option to keep derivations

### ISSUE THREAD

**Issue by nrdxp** [nlewo/nix2container#63]
There are a variety of Nix commands that can be run on derivation files themselves to avoid additional evaluation.

For example, if one calls `nix print-dev-env` on an outpath whose derivation exists in the store, then that derivation will be used as the basis for the shell that is loaded. If the derivation does not exist in the store the command fails. I would like to use this command to enter a devshell in a container without any extra evaluation (the devshell outpath already exists in the image), but I cannot do this because the derivation path is stripped out.

I also cannot reference `drvPath` directly as it adds over 20G of dependencies to the images, which is obviously not cool. Instead, It'd be nice to give users the option to copy over the derivation files for the outpaths included in the image.

**Comment by nlewo** [comment]
IIUC, in the `/nix/store` presents in the container, you would like to have the container derivation file, without its derivation file dependencies. I think that would be possible but this would lead to a broken `/nix/store` and i don't think that's a good idea.

**Comment by blaggacao** [comment]
![Untitled Diagram (1)](https://user-images.githubusercontent.com/7548295/214679364-88d994eb-8f72-4292-a885-9ae8f806d6d6.png)
> IIUC

Pretty much! "Kill the toil..."

**Comment by nlewo** [comment]
Hm, ok.

I think the [`closure-graph.json`](https://github.com/nlewo/nix2container/blob/6800fff45afecc7e47c334d14cf2b2f4f25601a0/default.nix#L226) file should contains both the store path and the drv paths. I however don't have any idea on how to do that :/

Could be related to this issue: https://github.com/NixOS/nix/issues/1245#issuecomment-401642781

**Comment by nrdxp** [comment]
> IIUC, in the `/nix/store` presents in the container, you would like to have the container derivation file, without its derivation file dependencies. I think that would be possible but this would lead to a broken `/nix/store` and i don't think that's a good idea.

No I don't mind having all the drv dependencies, they are usually pretty light since they are just text and maybe a few fetched sources. I think it'd be good to have as a non-default option.

**Comment by nlewo** [comment]
@nrdxp yes, I agree (and i have understood that with the @blaggacao diagram).
However, i think having runtime drvs without build time storepaths seems to be pretty difficult (i dont really understand all of these internal Nix mechanisms).

If the implementation is straightforward, i agree having an option would be nice.

**Comment by nrdxp** [comment]
I know how to do it in straight Nix. `nix copy` can do it but its kinda expensive as it copies one file at a time. The most efficient way to do it is with `nix-store --export` piped to a `nix-store --import`. Here is an example of how we do it in [std-action](https://github.com/divnix/std-action/blob/main/run/action.yaml#L116-L122) to pass the derivations to the runner jobs for them to build.

**Comment by nlewo** [comment]
@nrdxp yes, but this would have to be done at build time (in the sandbox), and i think that's much more difficult than doing it at runtime where you can query the Nix database.
